File: apps/habit-tracker/api/Models/Habit.cs
using System;
using System.Collections.Generic;

namespace HabitTracker.Api.Models
{
    public class Habit
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public AppUser? User { get; set; }
        public string Name { get; set; } = "";
        public string Icon { get; set; } = "target";
        public decimal TargetAmount { get; set; }
        public string Unit { get; set; } = "times";
        public string Frequency { get; set; } = "daily";
        public string Color { get; set; } = "#2563eb";
        public DateTimeOffset CreatedAt { get; set; }
        public ICollection<HabitEntry> Entries { get; set; } = new List<HabitEntry>();
    }
}

File: apps/habit-tracker/web/src/types/habit.ts
export type HabitFrequency = "daily" | "weekly" | "monthly";

export type Habit = {
  id: string;
  name: string;
  icon: string;
  targetAmount: number;
  unit: string;
  frequency: HabitFrequency;
  color: string;
  createdAt: string;
};

export type HabitEntry = {
  id: string;
  habitId: string;
  date: string;
  amount: number;
  note?: string;
};

export type HabitDraft = {
  name: string;
  targetAmount: number;
  unit: string;
  frequency: HabitFrequency;
  icon?: string;
  color: string;
};

export type CurrentUser = {
  id: string;
  displayName: string;
  email: string;
  avatarUrl?: string;
};
